### DISCUSSION TITLE
Data modelling question

### DISCUSSION THREAD

**Discussion by keesvanbemmel** [directus/directus#10425] [Archive]
Hi all!

Say I have a collection of Brands (Ferrari, Lamborghini, Skoda), a collection of Car categories (sports, family) and a collection of suppliers: (dealer a, dealer b).

I would like to store two things:
- For some combinations of brand & category (let's say a sports car of brand Ferrari) who the preferred supplier is (dealer b)
- For some combinations of brand & category (let's say a sports car of brand Ferrari) if we should promote them on our front page (check out our collection of Ferrari sportcars!)

What would be the most efficient way to store this?

Cheers!

**Comment by rijkvanzanten** [comment]
Great question. In this case, you'd use a many to many from both brands _and_ categories to the related _suppliers_, using the same junction table between. That will effectively create a new "preferred" table where you can lookup the preferred supplier for any given combination of category and brand. 

Visualized:

<img width="824" alt="CleanShot 2021-12-14 at 11 15 46@2x" src="https://user-images.githubusercontent.com/9141017/146036664-3f03672c-6b8c-4e91-aa30-9479962c41c5.png">

Another nice benefit of this is that you can add additional fields to this "preferred" table, like "Featured" or "Note" or any other content you might need that's specific to the combination of brand+category+supplier.

**Reply by keesvanbemmel** [reply]
Thanks @rijkvanzanten 

Follow up Q: what tool did you use to draw that model?

**Reply by paescuj** [reply]
Looks like [excalidraw](https://github.com/excalidraw/excalidraw) which I can highly recommend 😃

**Reply by rijkvanzanten** [reply]
That's exactly what it is 🙂 Love it, it's perfect for these quick brain-fart sketches

**Reply by keesvanbemmel** [reply]
Now the million dollar question: 

Let's say I have products that have relations to a brand and a category.

How can I query products and also return their preferred supplier in your data model @rijkvanzanten :)

**Reply by rijkvanzanten** [reply]
I think you'll have to fetch the product first, and then make a secondary query to preferred_suppliers with a filter to query where categories_id and brand_id match the id's in your original product. I don't think you can get the data you're looking for nested in a single request currently, as the nested loading for the product->category->preferred_supplier will fetch the nested preferred supplier for the category, but won't take the brand requirement into account.

Alternatively, you could potentially fetch the preferred supplier nested through either category or brand, and then use `deep` with a filter to account for the other side. However, in order to do that dynamically, you would need to rely on a way to dynamically inject a value from the parent record, which is not currently a supported feature (see https://github.com/directus/directus/discussions/2993 for more on that)
